### PR TITLE
Robotics/Science Power Tools

### PR DESCRIPTION
**Pull request by Youtubeboy139**
## About The Pull Request
Adds in a set of Science specific power tools that are identical to their industrial counterparts in function but lack the same speed and usability - notably the "powered crowbar" lacks the power to force open doors.

At the moment these are only Admin spawn items pending review of this PR. I don't want to do all this work and have the PR shot down last second for whatever reason. Will amend PR with in-game spawnability once approved by staff and maintainers.

## How This Contributes To The Skyrat Roleplay Experience
Gives Robotics and Science their own power tools so they no longer have to ask or in some cases pay Engineering for power tools and removes the risk of random players getting their hands on the coveted Jaws of Life and breaking into parts of the station they shouldn't have access to.

## Changelog

:cl:
add: Added new science power tools
imageadd: Added required icons for the new science power tools
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
Science powertool sprites already exist, in our modular tool icon file, and at a quality equal to engineering tools' sprites. I'd much rather we just use those instead of making new sprites that don't match existing sprites.

File: modular_skyrat/modules/sci_tool/crowbar.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * Signal proc for [COMSIG_TRANSFORMING_ON_TRANSFORM].
+ *
+ * Toggles between crowbar and wirecutters and gives feedback to the user.
+ */
+/obj/item/crowbar/science/proc/on_transform(obj/item/source, mob/user, active)
+	SIGNAL_HANDLER
+	tool_behaviour = (active ? TOOL_WIRECUTTER : TOOL_CROWBAR)
+	balloon_alert(user, "attached [active ? "cutting" : "prying"]")
+	playsound(user ? user : src, 'sound/items/change_drill.ogg', 50, TRUE)
+	return COMPONENT_NO_DEFAULT_MESSAGE
```
Comment (Zonespace27) [inline modular_skyrat/modules/sci_tool/crowbar.dm:45]: needs a newline

File: modular_skyrat/modules/sci_tool/crowbar.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/obj/item/crowbar/science
```
Comment (Zonespace27) [inline modular_skyrat/modules/sci_tool/crowbar.dm:1]: should be a subtype of jaws of life

File: modular_skyrat/modules/sci_tool/screwdriver.dm [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/obj/item/screwdriver/science
```
Comment (Zonespace27) [inline modular_skyrat/modules/sci_tool/screwdriver.dm:1]: should be a drill subtype

File: modular_skyrat/modules/sci_tool/screwdriver.dm [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	SIGNAL_HANDLER
+
+	tool_behaviour = (active ? TOOL_WRENCH : TOOL_SCREWDRIVER)
+	balloon_alert(user, "attached [active ? "bolt bit" : "screw bit"]")
+	playsound(user ? user : src, 'sound/items/change_drill.ogg', 50, TRUE)
+	return COMPONENT_NO_DEFAULT_MESSAGE
+
+/obj/item/screwdriver/science/examine()
+	. = ..()
+	. += " It's fitted with a [tool_behaviour == TOOL_SCREWDRIVER ? "screw" : "bolt"] bit."
```
Comment (Zonespace27) [inline modular_skyrat/modules/sci_tool/screwdriver.dm:53]: no newline

File: modular_skyrat/modules/sci_tool/screwdriver.dm [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		throwforce_on = throwforce, \
+		hitsound_on = hitsound, \
+		w_class_on = w_class, \
+		clumsy_check = FALSE)
+	RegisterSignal(src, COMSIG_TRANSFORMING_ON_TRANSFORM, .proc/on_transform)
+
+/*
+ * Signal proc for [COMSIG_TRANSFORMING_ON_TRANSFORM].
+ *
+ * Toggles between crowbar and wirecutters and gives feedback to the user.
```
Comment (Zonespace27) [inline modular_skyrat/modules/sci_tool/screwdriver.dm:41]: copy paste comments

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Not sold on the idea.

**Comment by Zonespace27** [comment]
> Gives Robotics and Science their own power tools so they no longer have to ask or in some cases pay Engineering for power tools

Why is this a bad thing; we want to INCREASE player interaction, not the opposite

**Comment by Elli-Skala** [comment]
> > Gives Robotics and Science their own power tools so they no longer have to ask or in some cases pay Engineering for power tools
> 
> Why is this a bad thing; we want to INCREASE player interaction, not the opposite

I mean to be fair with a autolathe and 30 seconds of time rnd can print power tools too

**Comment by Zonespace27** [comment]
> > > Gives Robotics and Science their own power tools so they no longer have to ask or in some cases pay Engineering for power tools
> > 
> > 
> > Why is this a bad thing; we want to INCREASE player interaction, not the opposite
> 
> I mean to be fair with a autolathe and 30 seconds of time rnd can print power tools too

Not jaws, only screwdriver + welder
Also making it even easier still isn't appealing

**Comment by GuiltyNeko** [comment]
As a CE main, (and for clarity, the person who bitched about this enough for this PR to get made) the problem with jaws of life is that it's objectively superior to normal tools in two different ways at the exact same time, which allows someone to claim that they want it to save toolbelt space, when in reality they actually want it to be able to force doors open.

Adjacent to that, there are some people in science who act as though they are entitled to jaws of life for having researched them, "I did you a favor, now you owe me" sort of attitude. This is obviously contradictory to the fact that jaws of life are contraband for those who aren't in engineering or a paramedic, as in, they have a legitimate reason to force open doors in carrying out their job. The problem with contraband is that I have not once heard of security arresting someone on skyrat for even so much as breaking into tech storage and stealing the insulated gloves, let alone having jaws of life when they aren't meant to.

At this point the PR was closed and deleted while I was typing this up so I guess I'm done.

**Pull request closed without merging**